variables.tf

```diff
@@ -9,3 +13,257 @@ variable "tags" {
+  type        = string
+}
+
+variable "asg_max_size" {
+  default     = 2
+  description = "Maximum number of instances in the autoscaling group"
+  type        = number
+}
+
+variable "asg_min_size" {
+  default     = 1
+  description = "Minimum number of instances in the autoscaling group"
+  type        = number
+}
+
+variable "asg_root_volume_size" {
+  default     = 20
+  description = "size of root volume (includes app install but not data dir)"
+  type        = number
+}
+
+variable "asg_subnet" {
+  description = "Subnet to associate ASG instances with (specify no more than 1)"
+  type        = string
+}
+
+#################################################
+# DB Settings
+#################################################
+
+variable "db_additional_security_groups" {
+  default     = []
+  description = "SGs permitted access to RDS"
+  type        = list(string)
+}
+
+variable "db_allowed_access_cidrs" {
+  default     = []
+  description = "CIDRs permitted access to RDS"
+  type        = list(string)
+}
+
+variable "db_engine_version" {
+  default     = "11"
+  description = "engine version to run"
+  type        = string
+}
+
+variable "db_storage_size" {
+  description = "Size in DB (in GB)"
+  type        = number
+}
+
+variable "db_instance_class" {
+  description = "DB Instance Size"
+  type        = string
+}
+
+variable "db_multi_az" {
+  default     = false
+  description = "If true, DB will be configured in multi-AZ mode"
+  type        = bool
+}
+
+variable "db_monitoring_role_arn" {
+  default     = null
+  description = "ARN for Database Monitoring (required for performance insights)"
+  type        = string
+}
+
+variable "db_monitoring_interval" {
+  default     = 0
+  description = "Enhanced monitoring interval (5-60 seconds, 0 to disable)"
+  type        = number
+}
+
+variable "db_parameters" {
+  description = "DB parameters (downstream module defaults will be used if not specified)"
+
+  default = [
+    {
+      name  = "client_encoding"
+      value = "UTF8"
+    }
+  ]
+
+  type = list(object({
+    name  = string
+    value = string
+  }))
+}
+
+variable "db_performance_insights_enabled" {
+  default     = false
+  description = "Whether or not to enable DB performance insights"
+  type        = bool
+}
+
+variable "db_subnet_group" {
+  description = "Database subnet group"
+  type        = string
+}
+
+variable "db_vpc_id" {
+  default     = null
+  description = "VPC ID for database (if omitted, the value for `vpc_id` is used instead)"
+  type        = string
+}
+
+########################################
+# Networking Vars
+########################################
+
+variable "availability_zone" {
+  description = "Specify the availability zone that the instance will be deployed in. Because Bitbucket requires an EBS volume for data and can't use EFS, the value of `availability_zone` must match the AZ associated with the value of `asg_subnet`."
+  type        = string
+}
```

Comment:
Why not just remove `var.asg_subnet` entirely and use this for both?

Reply:
An AZ can have more than one subnet. Usually not, but we have clients who undersized their subnets that have had to do this. Picking one seems less than ideal.